General improvements and bug fixes

digaru19 opened the pull request.

digaru19 committed:
Add global DEBUG flag

digaru19 committed:
Default behaviour to resolve all links; Fixes #20

digaru19 committed:
Fix linking of sections; Fixes #13

digaru19 committed:
Fix rendering of DTypes in linked sections; Fix drop at last row

achilleas-k commented:
Is there a way to toggle the `DEBUG` flag externally or can it only be done in code for now?

jgrewe commented:
maybe use argparse package to define the user interface

achilleas-k commented on odmlui/dnd/tree.py:
```diff
@@ -21,11 +21,11 @@ def move(self):
     @property
     def copy(self):
-        return (self.action & gtk.gdk.ACTION_COPY) != 0
+        return (self.action == gtk.gdk.ACTION_COPY) != 0
```
Clean up `!= 0`

achilleas-k commented on odmlui/dnd/tree.py:
```diff
@@ -21,11 +21,11 @@ def move(self):
     @property
     def link(self):
-        return (self.action & gtk.gdk.ACTION_LINK) != 0
+        return (self.action == gtk.gdk.ACTION_LINK) != 0
```
Clean up `!= 0`

achilleas-k reviewed (changes requested):
Some comments:
- Clean up the two `!= 0` for the action comparisons.
- Add a command line debug flag that sets `DEBUG = True`.
- Print errors to `stderr` and debug output to `stdout`.

digaru19 committed:
Simplify DType import (Fixes #24)

digaru19 committed:
Attach FileChooserDialog to parent window

digaru19 committed:
Fix Python 2 unicode display (Fixes #26)

digaru19 committed:
Display Error Dialog on opening invalid odML files

If odML file is malformed, display an ErrorDialog
(Fixes #27, Fixes #28)
Check if file exists before adding it to 'Recently used files' list
(Fixes #25)

digaru19 committed:
Add command line arguments '--files' and '--debug'

digaru19 committed:
Fix 'value' rendering in python 2; Disable tooltip in PropertyView

digaru19 committed:
Enable relative path resolution for file name

digaru19 committed:
Fix 'unit' rendering on edit

digaru19 committed:
Simplify 'DType' rendering for linked sections
